### PR TITLE
Updated `forceCast` implementation

### LINKED ISSUE #122: How to collect all cast error including unique constrain violations?
Validation for unique constraints doesn't appear to be [implemented](https://github.com/frictionlessdata/tableschema-js/blob/master/src/constraints/unique.js#L4)?
Could this be added to upcoming release or is the validation performed elsewhere?

### PR DESCRIPTION
**Pull request by roll**
- fixes #122 
- close #124

### CODE REVIEW

**Comment by roll** [comment]
@mattRedBox 
Please take a look. I wasn't able to PR directly to your branch so pushed it as a separate PR. I've updated the implementation slightly and added a test.

So for a client it will be simple:

```javascript
const table = await Table.load(source, {schema})
const rows = await table.read({forceCast: true})
for (const row of rows) {
  if (row.errors) // iterate over cast error
  // row is valid, handle an array
}
```

**Comment by Stephen-Gates** [comment]
Thanks @roll

**Comment by roll** [comment]
@Stephen-Gates 
Thanks!

@mattRedBox 
Could you please check (if needed) will it be enough to resolve the problem with `unique-constraint` (#122) we had at the first place?

File: src/table.js [hunk 1]
```diff
@@ -100,10 +100,11 @@ class Table {
       if (cast) {
         if (this.schema) {
           try {
-            row = this.schema.castRow(row)
+            row = this.schema.castRow(row, {failFast: false})
           } catch (error) {
             error.rowNumber = rowNumber
             error.errors.forEach(error => {error.rowNumber = rowNumber})
+            if (forceCast) return error
```
Comment (roll) [inline src/table.js]: TODO: return error for unique check fail also

**Comment by roll** [comment]
Released as https://github.com/frictionlessdata/tableschema-js/releases/tag/v1.7.0

**Comment by mattRedBox** [comment]
thx @roll,
haven't had a chance to test yet, but will let you know

**Comment by mattRedBox** [comment]
Thanks @roll.
Works very well.

**Comment by roll** [comment]
@mattRedBox 
Thanks for the feedback and PR!

**Pull request merged**